typed/typed.go

```diff
@@ -140,7 +140,13 @@ func (tv TypedValue) Compare(rhs *TypedValue) (c *Comparison, err error) {
 
 // RemoveItems removes each provided list or map item from the value.
 func (tv TypedValue) RemoveItems(items *fieldpath.Set) *TypedValue {
-	tv.value = removeItemsWithSchema(tv.value, items, tv.schema, tv.typeRef)
+	tv.value = removeItemsWithSchema(tv.value, items, tv.schema, tv.typeRef, false)
+	return &tv
+}
+
+// Remove removes each provided path from the value.
+func (tv TypedValue) Remove(paths *fieldpath.Set) *TypedValue {
```

Comment:
I don't know if that name is explicit enough (nor is the comment/doc above).

I would suggest `RemoveDeep()`, which will sound counter-intuitive to you, since you used shallow where I would have used `deep` :-))

Reply:
I'm okay with using deep instead if shallow.
Thought about it and deep didn't feel right looking at what it does, but I'm okay with it any way.

Reply:
Shallow and Deep are both used for copy, comparison, etc. I think it'll be idiomatic for people here too.